client/platform/desktop/backend/native/viame.ts

```diff
@@ -0,0 +1,248 @@
+import npath from 'path';
+import { spawn } from 'child_process';
+import fs from 'fs-extra';
+
+import {
+  Settings, DesktopJob, DesktopJobUpdate, RunPipeline, RunTraining,
+} from 'platform/desktop/constants';
+import { serialize } from 'platform/desktop/backend/serializers/viame';
+
+import * as common from './common';
+import { cleanString, jobFileEchoMiddleware } from './utils';
+
+const PipelineRelativeDir = 'configs/pipelines';
+
+interface ViameConstants {
+  setupScriptAbs: string; // abs path setup comman
+  trainingExe: string; // name of training binary on PATH
+  kwiverExe: string; // name of kwiver binary on PATH
+  shell: string | boolean; // shell arg for spawn
+}
+
+/**
+ * a node.js implementation of viame_tasks.tasks.run_pipeline
+ */
+async function runPipeline(
+  settings: Settings,
+  runPipelineArgs: RunPipeline,
+  updater: (msg: DesktopJobUpdate) => void,
+  validateViamePath: (settings: Settings) => Promise<true | string>,
+  viameConstants: ViameConstants,
+): Promise<DesktopJob> {
+  const { datasetId, pipeline } = runPipelineArgs;
+
+  const isValid = await validateViamePath(settings);
+  if (isValid !== true) {
+    throw new Error(isValid);
+  }
+
+  const pipelinePath = npath.join(settings.viamePath, PipelineRelativeDir, pipeline.pipe);
+  const projectInfo = await common.getValidatedProjectDir(settings, datasetId);
+  const meta = await common.loadJsonMetadata(projectInfo.metaFileAbsPath);
+  const jobWorkDir = await common.createKwiverRunWorkingDir(settings, [meta], pipeline.name);
+
+  const detectorOutput = npath.join(jobWorkDir, 'detector_output.csv');
+  const trackOutput = npath.join(jobWorkDir, 'track_output.csv');
+  const joblog = npath.join(jobWorkDir, 'runlog.txt');
+
+  let command: string[] = [];
+  if (meta.type === 'video') {
+    const videoAbsPath = npath.join(meta.originalBasePath, meta.originalVideoFile);
+    command = [
+      `${viameConstants.setupScriptAbs} &&`,
+      `${viameConstants.kwiverExe} runner`,
+      '-s input:video_reader:type=vidl_ffmpeg',
+      `-p ${pipelinePath}`,
+      `-s input:video_filename=${videoAbsPath}`,
+      `-s detector_writer:file_name=${detectorOutput}`,
+      `-s track_writer:file_name=${trackOutput}`,
```

Comment:
file paths need to be quoted for windows because they could have spaces in them.  So all paths need a surrounding quote to work.